### PR TITLE
Add support for JPEG to gdal2tiles.py

### LINKED ISSUE #6703: Feature Request for Gdal2Tiles: Support JPEG format
## Expected behavior and actual behavior.

Use gdal2tiles to create tiles in jpg format 👍 (tenth of the PNG size )
gdal2tiles currently supports only PNG or WEBP 😢 
https://gdal.org/programs/gdal2tiles.html#cmdoption-gdal2tiles-tiledriver

### PR DESCRIPTION
**Pull request by zdila**
## What does this PR do?

Add support for JPEG for `gdal2tiles.py`. Should fix #6703.

## Tasklist

 - [ ] ADD YOUR TASKS HERE
 - [ ] Add test case(s)
 - [ ] Add documentation
 - [ ] Updated Python API documentation (swig/include/python/docs/)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: swig/python/gdal-utils/osgeo_utils/gdal2tiles.py [hunk 1]
```diff
@@ -1838,6 +1859,17 @@ def optparse_init() -> optparse.OptionParser:
...
     p.add_option_group(g)
 
+    # Jpeg options
+    g = optparse.OptionGroup(p, "JPEG options", "Options for JPEG tiledriver")
+    g.add_option(
+        "--jpeg-quality",
+        dest="jpeg_quality",
+        type=int,
+        default=80,
+        help="quality of jpeg image, integer between 1 and 100, default is 75",
```
Comment (jjimenezshaw) [inline swig/python/gdal-utils/osgeo_utils/gdal2tiles.py:1869]: Default is 75 or 80?

Comment (asafMasa) [reply swig/python/gdal-utils/osgeo_utils/gdal2tiles.py:1869]: @jjimenezshaw @zdila is there any progress with this issue?

Comment (asafMasa) [reply swig/python/gdal-utils/osgeo_utils/gdal2tiles.py:1869]: By the way is there an option for creating "mixed" tiles (png in the borders and jpeg in the middle) as is done when using GDAL to create GPKG file?

Comment (jjimenezshaw) [reply swig/python/gdal-utils/osgeo_utils/gdal2tiles.py:1869]: > Default is 75 or 80?

My comment was just about a discrepancy between the default value and the documentation

**Comment by rouault** [comment]
I've continued this work in https://github.com/OSGeo/gdal/pull/9690 , fixing in particular overview tile generation, and adding tests. Hence closing this PR

**Pull request closed without merging**